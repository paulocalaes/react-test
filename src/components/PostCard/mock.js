export const postCardMock = {
  title: "title",
  body: "body",
  cover: "text",
  id: "1",
};
