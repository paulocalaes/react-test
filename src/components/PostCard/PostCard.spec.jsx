import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardMock } from "./mock";

const props = postCardMock;

describe("<Postcard />", () => {
  it("should render Postcard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: "title" })).toHaveAttribute(
      "src",
      "text"
    );

    expect(
      screen.getByRole("heading", { name: "title 1" })
    ).toBeInTheDocument();

    expect(screen.getByText("body")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
