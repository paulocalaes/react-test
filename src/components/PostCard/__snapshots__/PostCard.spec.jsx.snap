// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Postcard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title"
    src="text"
  />
  <div
    class="post-content"
  >
    <h2>
      title
       
      1
    </h2>
    <p>
      body
    </p>
  </div>
</div>
`;
